course5/lec5_en.md

```diff
@@ -0,0 +1,347 @@
+---
+marp: true
+math: mathjax
+paginate: true
+backgroundImage: url('../pics/background_moonbit.png')
+headingDivider: 1
+---
+
+# Modern Programming Ideology
+
+## Trees
+
+### MoonBit Open Course Team
+
+# Data Structures: Trees
+
+- Trees
+- Binary trees
+- Binary search trees
+- Binary balanced trees
+
+# Trees in Everyday Life
+
+- Many data structures in everyday life resemble a tree
+  - Pedigree chart (also known as, family tree)
+  - File structure
+  - Mathematical expressions
+  - ...
+
+![height:300px](../pics/trees.drawio.png)
+
+# Logical Structure of Trees
+
+- In data structures, a **tree** is a finite collection of **nodes** that have a hierarchical relationship
+  - Nodes are structures that store data, and there exists a parent-child relationship between them: parent nodes and child nodes
+  - If the tree is not empty, it has a **root node**, which has no parent node
+  - All non-root nodes have a unique parent node
+  - Nodes without child nodes can be called leaf nodes
+  - No node can be its own descendant: the tree cannot have cycles
+  - An edge in a tree refers to a pair of nodes (u, v), where u is v's parent or v is u's parent
+
+![](../pics/abstract-tree.drawio.png)
+
+# Logical Structure of Trees
+
+- This is not a tree
+  
+![](../pics/not-a-tree.drawio.png)
```

Comment:
Could you please translate them and export to webp?

Reply:
Please check [63161cf](https://github.com/moonbitlang/moonbit-course/pull/15/commits/63161cfd96481f4b1b6c5c308e3d9c95e206f4e4)